utils/misc/asyncinit.go

```diff
@@ -0,0 +1,54 @@
+package misc
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+)
+
+// NewAsyncInit returns a new AsyncInit object with the given expected initialization events count.
+func NewAsyncInit(count int64) *AsyncInit {
+	a := &AsyncInit{
+		c: make(chan struct{}),
+	}
+	a.count.Store(count)
+	return a
+}
+
+// AsyncInit is a helper object to wait for multiple asynchronous initialization events.
+type AsyncInit struct {
```

Comment:
It looks like `WaitContext` is used in tests only. Why not use a `WaitGroup` instead of creating this new `AsyncInit` type?

Reply:
Using a wait group was my initial approach, but `WaitGroup#Wait` is a blocking call, whereas listening from a channel doesn't have to be. Consider this as a `WaitGroup` using a channel for communicating completion instead of blocking.

Reply:
Makes sense, it's not too complex either so :+1: